### ISSUE TITLE
Documentation needs to be clearer about consumption

### ISSUE THREAD

**Issue by kentfredric** [chocolateboy/autobox#3]
I was talking to somebody today on IRC, who was trying to use autobox and was being confused why none of the code in the synopsis as-is.

I think the synopsis and documentation needs to be more explict in that `autobox.pm` creates none of the methods, and `autobox.pm` is explicitly a toolkit for other autoboxing things to use, and that the synopsis is only an example of what *could* be done once all the relevant subs are installed in `SCALAR::length()` etc, and that unless somebody is explicitly authoring a autoboxing module, that they should probably be consuming an existing autoboxer, like `autobox::Core` or `MooseX::Autobox`

**Comment by chocolateboy** [comment]
I totally agree. But I'm not gonna change the documentation because I can't think of a better way to express what autobox enables. If you can, please submit a PR.
